simulator/simulator.go

```diff
@@ -32,6 +34,9 @@ type HostSimulation struct {
 	//   h.state.lock.Lock()
 	//   defer h.state.lock.Unlock()
 	state *hostSimulationState
+
+	shutdown chan bool
```

Comment:
You can accomplish the shutdown with a single field, which I think is a bit cleaner: 

```
type Runner struct {
        shutdown   chan chan struct{}
}

func (r *Runner) loop() {
...
        case quit := <-r.shutdown:
            close(quit)
            return
        }   
}

func (r *Runner) Shutdown() {
    quit := make(chan struct{})
    r.shutdown <- quit
    <-quit
}
```